src/main.cc

```diff
@@ -106,6 +105,11 @@ static struct fiber *on_shutdown_fiber = NULL;
 static bool is_shutting_down = false;
 static int exit_code = 0;
 
+/* How many command line arguments we expect. */
+#define MAX_SAFE_ARGC 64
+/* Maximum valid command line argument length. */
+#define MAX_SAFE_ARGLEN 2048
```

Comment:
It'd be nice to have a comment explaining why we need to limit them.

Reply:
Done.